### PR TITLE
Create Single Game Window for Host and Join States

### PR DESCRIPTION
**Pull request by thsparks**
This PR is probably too big. If needed, let me know and I can break it up into smaller pieces.

The primary change here is to create a single "GamePage" to host the sim, players, and reaction content. The HostGame and JoinGame components have been repurposed/reduced into simple pages where a user can enter a game link or join code respectively. In the mid-to-long term, both HostGame and JoinGame should be entirely skippable with direct links (i.e. a "Host a Multiplayer Session for this Game" link would bypass HostGame, and a "Join Link" would bypass JoinGame. Both would go straight to the GamePage lobby state).

Meanwhile, the "lobby" (both join and host) is now represented with modals. They're not a perfect 1:1 with designs, but it's a start in that direction. Unclear if these will actually be modals in the long-term.

I've also done some initial UI work on the presence and reactions to bring them closer to the designs, but this isn't complete. It's still very rough around the edges, but this PR was already getting too big so I decided to send as-is, then we can finalize the UI piecemeal in later changes.

Some screenshots:
Game Page ("Keyboard Controls" doesn't work yet, just a placeholder):
image

Host Lobby (Note, we don't have a link to provide yet, so still just gives code for now):
image

Join Lobby:
image

Reaction Bar (WIP):
image

Empy Player Slots:
image

### CODE REVIEW

**Review by jwunderl — APPROVED** [review]

File: multiplayer/src/components/AppModal.tsx [hunk 1]
```diff
@@ -20,6 +22,10 @@ export default function Render() {
             return <Modal title={lf("Report Abuse")} onClose={() => dispatch(clearModal())}>
                 Report Abuse Placeholder  {/*TODO multiplayer*/}
             </Modal>
+        case "host-lobby":
+            return <HostLobbyModal />
+        case "join-lobby":
+            return <JoinLobbyModal />
         default:
             return <div />
```
Comment (eanders-ms) [inline multiplayer/src/components/AppModal.tsx:30]: It would be ok to return `null` here.

File: multiplayer/src/components/HostLobbyModal.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    let inputRef: HTMLInputElement;
+
+    const onStartGameClick = async () => {
+        pxt.tickEvent("mp.hostlobby.startgame");
+        dispatch(clearModal());
+        await startGameAsync();
+    };
+
+    const handleCopyClick = () => {
+        if (pxt.BrowserUtils.isIpcRenderer()) {
```
Comment (eanders-ms) [inline multiplayer/src/components/HostLobbyModal.tsx:21]: futureproofing. i like it. :)

File: multiplayer/src/components/HostLobbyModal.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import { Input } from "react-common/components/controls/Input";
+import { Modal } from "react-common/components/controls/Modal";
+import { startGameAsync } from "../epics";
+import { clearModal } from "../state/actions";
+import { AppStateContext } from "../state/AppStateContext";
+
+export default function Render() {
+    const { state, dispatch } = useContext(AppStateContext);
+    const [ copySuccessful, setCopySuccessful ] = useState(false);
+    let inputRef: HTMLInputElement;
```
Comment (eanders-ms) [inline multiplayer/src/components/HostLobbyModal.tsx:12]:
```suggestion
const inputRef = useRef<HTMLInputElement>(null);
```

File: multiplayer/src/components/HostLobbyModal.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    return (
+        <Modal title={lf("Invite Players")} onClose={handleOnClose}>
+            <div className="tw-flex tw-flex-col tw-gap-1 tw-items-center">
+                <div className="tw-flex tw-flex-col tw-items-center">
+                    <div>{lf("Invite anyone to join your game instantly.")}</div>
+                    <div className="tw-mt-1">{lf("Just send them a link.")}</div>
+                    <div className="common-input-attached-button tw-m-5 tw-w-full">
+                                <Input
+                                    ariaLabel={lf("join game link")}
+                                    handleInputRef={handleInputRef}
```
Comment (eanders-ms) [inline multiplayer/src/components/HostLobbyModal.tsx:54]:
```suggestion
                                    handleInputRef={inputRef}
```

File: multiplayer/src/components/SignedInPage.tsx [hunk 5]
```diff
@@ -5,25 +5,19 @@ import { signOutAsync } from "../epics";
...
-            <Button
-                className="primary"
-                label={authButtonLabel}
-                title={authButtonLabel}
-                onClick={signOutAsync}
-            />
             {uiMode === "home" && <JoinOrHost />}
-            {uiMode === "host" && <HostGame />}
-            {uiMode === "join" && <JoinGame />}
+            {netMode == "init" && uiMode === "host" && <HostGame />}
```
Comment (eanders-ms) [inline multiplayer/src/components/SignedInPage.tsx:18]:
```suggestion
            {netMode === "init" && uiMode === "host" && <HostGame />}
```

File: multiplayer/src/components/SignedInPage.tsx [hunk 6]
```diff
@@ -5,25 +5,19 @@ import { signOutAsync } from "../epics";
...
+            {netMode == "init" && uiMode === "join" && <JoinGame />}
```
Comment (eanders-ms) [inline multiplayer/src/components/SignedInPage.tsx:19]:
```suggestion
            {netMode === "init" && uiMode === "join" && <JoinGame />}
```

File: multiplayer/src/components/SignedInPage.tsx [hunk 7]
```diff
@@ -5,25 +5,19 @@ import { signOutAsync } from "../epics";
...
+            {(netMode == "connecting" || netMode == "connected") && <GamePage />}
```
Comment (eanders-ms) [inline multiplayer/src/components/SignedInPage.tsx:20]:
```suggestion
            {(netMode === "connecting" || netMode === "connected") && <GamePage />}
```
nit, just for consistency.

**Review by eanders-ms — APPROVED** [review]
Looks good! Made a few small suggestions.

File: multiplayer/src/components/HostLobbyModal.tsx (hunk 2 above)
Comment (thsparks) [reply multiplayer/src/components/HostLobbyModal.tsx:21]: This was all shamelessly copied from https://github.com/microsoft/pxt/blob/multiplayer/thsparks/unify_game_window/react-common/components/share/ShareInfo.tsx#L64-L72 :)

**Pull request merged**